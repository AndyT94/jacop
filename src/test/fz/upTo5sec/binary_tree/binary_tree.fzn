var 1..1: X_INTRODUCED_0;
var 3..3: X_INTRODUCED_1;
var 5..5: X_INTRODUCED_2;
var 7..7: X_INTRODUCED_3;
var 1..1: X_INTRODUCED_4;
var 3..3: X_INTRODUCED_5;
var 7..7: X_INTRODUCED_6;
var 5..5: X_INTRODUCED_7;
var 2..2: num_trees:: output_var = 2;
var 1..1: X_INTRODUCED_159 ::var_is_introduced  = 1;
var 5..5: X_INTRODUCED_161 ::var_is_introduced  = 5;
var 1..1: X_INTRODUCED_163 ::var_is_introduced  = 1;
var 7..7: X_INTRODUCED_165 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_167 ::var_is_introduced  = 1;
var 5..5: X_INTRODUCED_169 ::var_is_introduced  = 5;
var 7..7: X_INTRODUCED_171 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_173 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_175 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_177 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_179 ::var_is_introduced  = 1;
var 7..7: X_INTRODUCED_181 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_183 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_185 ::var_is_introduced  = 1;
var 7..7: X_INTRODUCED_187 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_189 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_191 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_193 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_195 ::var_is_introduced  = 1;
var 7..7: X_INTRODUCED_197 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_199 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_201 ::var_is_introduced  = 1;
var 7..7: X_INTRODUCED_203 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_205 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_207 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_209 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_211 ::var_is_introduced  = 1;
var 7..7: X_INTRODUCED_213 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_215 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_217 ::var_is_introduced  = 1;
var 7..7: X_INTRODUCED_219 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_221 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_223 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_225 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_227 ::var_is_introduced  = 1;
var 7..7: X_INTRODUCED_229 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_231 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_233 ::var_is_introduced  = 1;
var 7..7: X_INTRODUCED_235 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_237 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_239 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_241 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_243 ::var_is_introduced  = 1;
var 7..7: X_INTRODUCED_245 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_247 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_249 ::var_is_introduced  = 1;
var 7..7: X_INTRODUCED_251 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_253 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_255 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_257 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_259 ::var_is_introduced  = 1;
var 7..7: X_INTRODUCED_261 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_263 ::var_is_introduced  = 1;
var 1..1: X_INTRODUCED_265 ::var_is_introduced  = 1;
var 7..7: X_INTRODUCED_267 ::var_is_introduced  = 7;
var 1..1: X_INTRODUCED_269 ::var_is_introduced  = 1;
array [1..8] of var int: tree:: output_array([1..8]) = [1,3,5,7,1,3,7,5];
array [1..64] of var int: parents:: output_array([1..8,1..8]) = [1,3,5,7,1,3,7,5,1,5,1,7,1,5,7,1,1,1,1,7,1,1,7,1,1,1,1,7,1,1,7,1,1,1,1,7,1,1,7,1,1,1,1,7,1,1,7,1,1,1,1,7,1,1,7,1,1,1,1,7,1,1,7,1];
array [1..73] of var int: X_INTRODUCED_74 ::var_is_introduced  = [1,3,5,7,1,3,7,5,1,5,1,7,1,5,7,1,1,1,1,7,1,1,7,1,1,1,1,7,1,1,7,1,1,1,1,7,1,1,7,1,1,1,1,7,1,1,7,1,1,1,1,7,1,1,7,1,1,1,1,7,1,1,7,1,1,3,5,7,1,3,7,5,2];
solve :: int_search(X_INTRODUCED_74,first_fail,indomain_min,complete) satisfy;
