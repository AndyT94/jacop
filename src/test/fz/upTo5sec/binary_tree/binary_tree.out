num_trees = 2;
tree = array1d(1..8, [1, 3, 5, 7, 1, 3, 7, 5]);
parents = array2d(1..8, 1..8, [1, 3, 5, 7, 1, 3, 7, 5, 1, 5, 1, 7, 1, 5, 7, 1, 1, 1, 1, 7, 1, 1, 7, 1, 1, 1, 1, 7, 1, 1, 7, 1, 1, 1, 1, 7, 1, 1, 7, 1, 1, 1, 1, 7, 1, 1, 7, 1, 1, 1, 1, 7, 1, 1, 7, 1, 1, 1, 1, 7, 1, 1, 7, 1]);
----------